### DISCUSSION TITLE
ERROR: mount Dash app (Flask) at (home) "/" path

### DISCUSSION THREAD

**Discussion by iBuitron** [fastapi/fastapi#12608] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

I have two dash app
- Dash1
- Dash2

dash1/app.py

```python
from dash import Dash, html

app = Dash(__name__, requests_pathname_prefix="/dash1/")
app.layout = html.Div([html.H1("DASH 1 APP")])

if __name__ == "__main__":
    app.run_server(debug=True, port=8051)

```

dash2/app.py
```python
from dash import Dash, html
app = Dash(__name__, requests_pathname_prefix="/dash2/")
app.layout = html.Div([html.H1("DASH 2 APP")])

if __name__ == "__main__":
    app.run_server(debug=True, port=8052)

```

and a entrypoint 

app.py

```python
import uvicorn
from dash1 import app as dash1
from dash2 import app as dash2
from fastapi import FastAPI
from fastapi.middleware.wsgi import WSGIMiddleware

app = FastAPI()


@app.get("/")
def read_main():
    return {
        "message": "FastAPI ",
        "endpoints_disponibles": {"fastapi": "/", "dash1": "/dash1", "dash2": "/dash2"},
    }

app.mount("/dash1", WSGIMiddleware(dash1.app.server))
app.mount("/dash2", WSGIMiddleware(dash2.app.server))


if __name__ == "__main__":
    uvicorn.run(
        "app:app", host="0.0.0.0", port=8000, reload=True, log_level="info", workers=1
    )
```
All this works fine, but

## PROBLEM

I want to set dash1 at "/" (home),
when i change the requests_pathname_prefix from "/dash1/" to "/" (its a must in dash) 
and set it in` app.mount("/", WSGIMiddleware(dash1.app.server))`  and remove the @app.get("/")...

all breaks.

there is a way to set any mount app as  "home" ("/") ?

on this new config:

dash1/app.py

```python
from dash import Dash, html


app = Dash(__name__, requests_pathname_prefix="/")
app.layout = html.Div([html.H1("DASH 1 APP")])

if __name__ == "__main__":
    app.run_server(debug=True, port=8051)
```

(dash2 still the same)

entry / app.py
```python
import uvicorn
from dash1 import app as dash1
from dash2 import app as dash2
from fastapi import FastAPI
from fastapi.middleware.wsgi import WSGIMiddleware

app = FastAPI()


# @app.get("/")
# def read_main():
#     return {
#         "message": "FastAPI Principal AA",
#         "endpoints_disponibles": {"fastapi": "/", "dash1": "/dash1", "dash2": "/dash2"},
#     }


app.mount("/", WSGIMiddleware(dash1.app.server))
app.mount("/dash2", WSGIMiddleware(dash2.app.server))


if __name__ == "__main__":
    uvicorn.run(
        "app:app", host="0.0.0.0", port=8000, reload=True, log_level="info", workers=1
    )

```

Any path will redirecto to "/" 
![image](https://github.com/user-attachments/assets/8bc9e5e3-93b4-4d23-88a7-7b3814a0885b)
```


### Operating System

Linux, Windows

### Operating System Details

Ubuntu WSL2

### FastAPI Version

0.115.3

### Pydantic Version

2.9.2

### Python Version

Python 3.12.5

### Additional Context

_No response_

**Comment by iBuitron** [comment]
@tiangolo sry for tagged you, but there's any way?

**Comment by YuriiMotov** [comment]
What if you mount `dash2` first and then `dash1`?

```python
app.mount("/dash2", WSGIMiddleware(dash2.app.server))
app.mount("/", WSGIMiddleware(dash1.app.server))
```

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
